handler/NeighboursHandler.py

```diff
@@ -74,6 +80,11 @@ def serve(self, sd: socket.socket):
 
 		# log the request received
 		socket_ip_sender = sd.getpeername()[0]
+		if ipaddress.IPv6Address(socket_ip_sender).ipv4_mapped is None:
+			socket_ip_sender = ipaddress.IPv6Address(socket_ip_sender).compressed
+		else:
+			socket_ip_sender = ipaddress.IPv6Address(socket_ip_sender).ipv4_mapped.compressed
+
```

Comment:
non ho capito questo passaggio, se dopo fai un vocale in cui spieghi cosa fa questo if te ne sarò eternamente grato